### ISSUE TITLE
Using Picasso to load images resets RecyclerView position to 1

### ISSUE THREAD

**Issue by krishnakandula** [Cleveroad/FanLayoutManager#8]
I'm using Picasso to load images into my ImageView. When I scroll, the position of the RecyclerView constantly gets reset to 1. Here is my code setting up the RecyclerView and FanLayoutManager.
```
private void setupRecyclerView(){
        mPokemonListAdapter = new PokemonListAdapter(new ArrayList<PokemonListItem>(), getContext());

        FanLayoutManagerSettings fanLayoutManagerSettings = FanLayoutManagerSettings
                .newBuilder(getContext())
                .withFanRadius(true)
                .withAngleItemBounce(5)
                .withViewHeightDp(400)
                .withViewWidthDp(300)
                .build();

        FanLayoutManager fanLayoutManager = new FanLayoutManager(getContext(), fanLayoutManagerSettings);
        mPokemonRecyclerView.setLayoutManager(fanLayoutManager);
        mPokemonRecyclerView.setAdapter(mPokemonListAdapter);
        mPokemonRecyclerView.setHasFixedSize(true);
        mPokemonRecyclerView.setDrawingCacheEnabled(true);
        mPokemonRecyclerView.setItemViewCacheSize(30);
        mPokemonRecyclerView.setDrawingCacheQuality(View.DRAWING_CACHE_QUALITY_LOW);
    }
```
And here, I use Picasso in my ViewHolder:
```
class ListViewHolder extends RecyclerView.ViewHolder {
        private final String LOG_TAG = ListViewHolder.class.getSimpleName();
        TextView mIdTextView;
        TextView mNameTypeTextView;
        CircleImageView mArtworkImageView;
        CardView mCardView;
        TextView mDescriptionTextView;

        final String defaultBackgroundColor = "#FFFFFF";  //White: default CardView background color

        public ListViewHolder(View view) {
            super(view);
            mIdTextView = (TextView) view.findViewById(R.id.pokemon_list_item_id_textview);
            mNameTypeTextView = (TextView) view.findViewById(R.id.pokemon_list_item_name_type_textview);
            mArtworkImageView = (CircleImageView) view.findViewById(R.id.pokemon_list_item_artwork_imageview);
            mCardView = (CardView) view.findViewById(R.id.pokemon_list_item_cardview);
            mDescriptionTextView = (TextView) view.findViewById(R.id.pokemon_list_item_description_textview);
        }

        public void bind(@NonNull PokemonListItem pokemon) {
            mIdTextView.setText("#" + pokemon.getId());
            StringBuilder nameType = new StringBuilder(String.format("%s \n%s", pokemon.getName(), pokemon.getType1()));
            if (pokemon.getType2() != null)
                nameType.append(String.format("/%s", pokemon.getType2()));
            mNameTypeTextView.setText(nameType.toString());
            mDescriptionTextView.setText(pokemon.getDescription());
            Log.v(LOG_TAG, "Grabbing artwork.");
            PokemonRepositories.getInMemoryPokemonRepository()
                    .getArtworkUri(pokemon.getId(), new PokemonRepository.GetArtworkUriCallback() {
                        @Override
                        public void onArtworkUriLoaded(Uri uri) {
                            updateArtwork(uri);
                        }
                    });
        }

        public void updateArtwork(final Uri artworkUri) {
            Picasso.with(mContext).cancelRequest(mArtworkImageView);
            Picasso.with(mContext)
                    .load(artworkUri.toString())
                    .fit().centerCrop()
                    .noPlaceholder()
                    .into(mArtworkImageView, new Callback() {
                        @Override
                        public void onSuccess() {
                            //Update drawable color asynchronously
                            Bitmap imageBitmap = ((BitmapDrawable) mArtworkImageView.getDrawable()).getBitmap();
                            Palette.from(imageBitmap).generate(new Palette.PaletteAsyncListener() {
                                @Override
                                public void onGenerated(Palette palette) {
                                    mCardView.setBackgroundColor(palette.getMutedColor(Color.parseColor(defaultBackgroundColor)));
                                }
                            });
                        }

                        @Override
                        public void onError() {
                            String errorMessage = String.format("ERROR: Could not load image with URI %s", artworkUri);
                            Log.e(LOG_TAG, errorMessage);
                        }
                    });
        }
    }
```

**Comment by yarovoiag** [comment]
Which version of the library has been used in this application ?

**Comment by krishnakandula** [comment]
I'm using the latest version of both Picasso(v2.5.2) and FanLayoutManager(v1.0.4)

**Comment by lockeb** [comment]
I have observed the same behavior with Picasso and Glide. When scrolling through the list it resets back to the start.

It works great with Fresco.

**Comment by krishnakandula** [comment]
@blocke79 Thanks for the reply, I'll also try it out with Fresco.

**Comment by yarovoiag** [comment]
I will solve the problem soon. No panic )

**Comment by ghost** [comment]
Any news on that?

**Comment by kzotin** [comment]
+1 concerned in ETA.
For me, switching to Fresco made scrolling freezing :(

**Comment by yarovoiag** [comment]
Fixed in version 1.0.5 ... finally...
